### ISSUE TITLE
HTTP 401 in KlarnaEnvironment.TestingNorthAmerica results in HTML response that cannot be deserialized to ErrorMessage resulting in ambiguous error message

### ISSUE THREAD

**Issue by ctorx** [klarna/kco_rest_dotnet#44]
**Describe the bug**
When attempting to create a payment credit session using `Klarna.Payment.CreateCreditSession` for Environment `TestingNorthAmerica`, an unauthorized attempt (HTTP 401) results in a response containing HTML that is attempted to be deserialized to type ErrorMessage, resulting in a deserialization error that is ambiguous to the caller and gives no indication that the failure is related to authorization.

The message is: "Unexpected character encountered while parsing value: <. Path '', line 0, position 0."

**Which service do you use**
Klarna Payments

**To Reproduce**
1. Make a call to `Klarna.Payment.CreateCreditSession` using invalid credentials.
2. Note the exception that is thrown in `BaseRestClient.ThrowIfError()` on this line: `errorMessage = JsonConvert.DeserializeObject<ErrorMessage>(content);`

**Expected behavior**
An exception should be thrown that indicates a HTTP 401 - Unauthorized attempt has occurred.  This may very well be solvable by configuring the web server to return a valid JSON object that can be deserialized to type `ErrorMessage`.

**Comment by alexions** [comment]
Hi @ctorx 

Thanks for reporting the issue. Unfortunately, there is no chance to make a proper reply with a proper error message (like Unauthorized) for all the Klarna services, so we can consider this way as "unfixable".

But I totally agree with you, that there is no reason to return Json parse exception. So I created a fix for the issue:
https://github.com/klarna/kco_rest_dotnet/pull/48

The problem for SDK is - we got unexpected result, we failed to parse and convert it.
If so - just return the text to customer and let customer make a decision.

After the fix (see the Sample project) you will get something like:
```
Status code: Unauthorized
Error: <html>
<head><title>401 Authorization Required</title></head>
<body>
<center><h1>401 Authorization Required</h1></center>
<hr><center>openresty</center>
</body>
</html>
```

Feel free to review the PR and comment the change.
Thanks!
